shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.mm

```diff
@@ -342,3 +322,43 @@ GrYUVABackendTextures yuvaBackendTextures(yuvaInfo, skiaBackendTextures,
 #endif  //  SLIMPELLER
 }
 @end
+
+@implementation FlutterDarwinExternalTextureImpellerImageWrapper
+
++ (sk_sp<flutter::DlImage>)wrapYUVATexture:(id<MTLTexture>)yTex
+                                     UVTex:(id<MTLTexture>)uvTex
+                             YUVColorSpace:(impeller::YUVColorSpace)colorSpace
+                               aiksContext:(nonnull impeller::AiksContext*)aiks_context {
+  impeller::TextureDescriptor yDesc;
+  yDesc.storage_mode = impeller::StorageMode::kDevicePrivate;
+  yDesc.format = impeller::PixelFormat::kR8UNormInt;
+  yDesc.size = impeller::ISize(yTex.width, yTex.height);
```

Comment:
Note that this code was hitting asserts running the AV video player example app because the sizes supplied didn't match the actual texture sizes. I changed the logic to read the texture descriptor size from the textures themselves so this can't get out of sync.